webhook_server/libs/github_api.py

```
@@ -105,90 +118,218 @@ def __init__(self, hook_data: dict[Any, Any], headers: Headers, logger: logging.
+                    f"(rate limit reset occurred - initial: {self.initial_rate_limit_remaining}, "
+                    f"final: {final_remaining})"
+                )
+            else:
+                token_spend = self.initial_rate_limit_remaining - final_remaining
+                # Log token spend with structured format for parsing
+                self.logger.info(
+                    f"{self.log_prefix} Token spend: {token_spend} API calls "
+                    f"(initial: {self.initial_rate_limit_remaining}, "
+                    f"final: {final_remaining}, remaining: {final_remaining})"
+                )
+        except Exception as ex:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)